src/main/java/org/broadinstitute/hellbender/tools/exome/ExonDB.java

```diff
@@ -0,0 +1,370 @@
+     * @throws IndexOutOfBoundsException if {@code index} is not within
+     *                                   valid bounds <code>[0 .. ({@link #exonCount()})</code>.
+     */
+    E exon(final int index);
+
+    /**
+     * Returns the exon given its name.
+     *
+     * @param name the query exon name.
+     * @return {@code null} if there is not such an exon.
+     */
+    default E exon(final String name) {
+        final int index = index(name);
+        if (index < 0)
+            return null;
+        else
+            return exon(index);
+    }
+
+    /**
+     * Returns the exon index given its name.
+     *
+     * @param name the query exon name.
+     * @return {@code null} if there is not such an exon.
+     */
+    int index(final String name);
+
+    /**
+     * Returns the exon that overlap a particular location.
+     * <p>The returned exon coordinates does not need to match exactly the
+     * the query {@code location} just overlap it by at least one base</p>.
+     * <p>This method will return {@code null} if there is no such a exon.
+     * </p>
+     *
+     * @param overlapRegion the query location.
+     * @return {@code null} if there is no overlapping exons at {@code overlapRegion}.
+     * @throws NullPointerException   if {@code overlapRegion} is {@code null}.
+     * @throws AmbiguousExonException if the query location overlaps more than one exon.
+     */
+    default E exon(final GenomeLoc overlapRegion) {
+        final int index = index(overlapRegion);
+        return index < 0 ? null : exon(index);
+
+    }
+
+    /**
+     * Returns the index of the exon that overlap a particular location.
+     * <p>The returned index's exon coordinates does not need to match exactly the
+     * the query {@code location} just overlap it by at least one base</p>.
+     * <p>This method will return a negative value if there is no such a exon.
+     *    This value can be transform to recover the position where the query location would be inserted
+     *    if it was present as <code>-(index(loc) + 1)</code>.
+     * </p>
+     *
+     * @param location the query location.
+     * @return less than 0 if there is no overlapping exons at {@code location}.
+     * @throws NullPointerException   if {@code location} is {@code null}.
+     * @throws AmbiguousExonException if the query location overlaps more than one exon.
+     */
+    int index(final GenomeLoc location);
+
+    /**
+     * Run a task (lambda function) on the set of exons that overlap a location.
+     */
+    default void forEachExon(final GenomeLoc overlapRegion, final ExonTask<E> exonTask) {
+        if (exonTask == null)
+            throw new NullPointerException();
+        final IndexRange indexRange = indexRange(overlapRegion);
+        for (int i = indexRange.from; i < indexRange.to; i++)
+            exonTask.run(i, exon(i));
+    }
+
+    /**
+     * Returns the exome-size in base-pairs as the sum of all its exon's sizes.
```

Comment:
`Returns the size of the exome in base-pairs (computed as the sum of all its exons sizes).`